CONTRIBUTING.md

```diff
@@ -0,0 +1,114 @@
+
+## Releasing (maintainers only)
+
+Before merging to main, the CI checks that all unit tests pass and that coverage does not decrease.  Merges should not occur until all checks pass.
+The Github testing workflow can be in [.github/workflows/main.yaml](.github/workflows/main.yaml) and is testing against Python version 3.10 and 3.12 on Ubuntu and Windows.
+The divergent changes to the `docs` branch should be pulled into your release branch as the [.github/workflows/docs.yaml](.github/workflows/docs.yaml) workflow is triggered by both pushes to main and docs.
+
+### Release notes and version
+On a release or feature branch, update the [CHANGELOG.md](./CHANGELOG.md) file with a terse list of changes for the release, oriented for users.  We organize changes by 'added' - new functions/modules/packages; 'modified' - changes to current functions and modules; 'removed' - deleted functions/modules/packages.  A short one or two sentence description below the version heading should describe the principle changes.  For example:
+
+```markdown
+## [Latest](https://github.com/int-brain-lab/ONE/commits/main) [X.X.X]
+Short sentence describing major changes.
+
+### Modified
+
+- ...
+
+### Added
+
+- ...
+
+### Removed
+
+- ...
+
+```
+
+The ONE version should be iterated. The version is defined in [one/\_\_init\_\_.py](./one/__init__.py) and follows the
+[Python packaging semver specification](https://packaging.python.org/en/latest/specifications/version-specifiers/#version-specifiers).
+
+* Patches (X.X.1) are small or urgent bug fixes or changes that don't affect compatibility.
+* Minor releases (X.1.X) are new features such as added functions or small changes that don't cause major compatibility issues.
+* Major releases (1.X.X) are major new features or changes that break backward compatibility in a big way.
+
+### Releasing to PyPi
+
+Once merged to main you can make a release to PyPi using the [.github/workflows/python-publish.yaml](.github/workflows/python-publish.yaml) workflow:
+
+1. Draft a [release on Github](https://github.com/int-brain-lab/ONE/releases).
+2. Title the release as the version number, starting with 'v'.
+3. Copy and paste the release notes from [CHANGELOG.md](./CHANGELOG.md).
+4. Create a tag with the version. Note that in order to trigger a PyPi release the tag must begin with 'v', e.g. `v2.8.0`.
+
+To release from Git:
+> [!WARNING]  
+> Releasing via git is not recormmended as the CI checks may be bypassed.
+
+```shell
+git checkout -b release/X.X.X origin/<branch>
+git checkout origin/main
+git merge release/X.X.X
+git tag vX.X.X
+git push origin --tags
+git push origin
+git branch -d release/X.X.X
+```
+
+To release without Github:
+> [!WARNING]  
+> Releasing to PyPi directly is not recommended as this may cause conflicts with the Github action.
```

Comment:
same here, if we settle on github action pypi upload, then there is no need to explain how to do it wrong